packages/connect-query/src/jest/test-utils.tsx

```diff
@@ -139,6 +140,24 @@ export const mockStatefulBigIntTransport = () =>
     });
   });
 
+/**
+ * a mock for PaginatedService that acts as an impromptu database
+ */
+export const mockStatefulPaginatedTransport = () =>
+  createRouterTransport(({ service }) => {
+    let base = 1n;
+    service(PaginatedService, {
+      list: (request) => {
+        const result = {
+          page: request.page,
+          items: [`${base} Item`, `${base + 1n} Item`, `${base + 2n} Item`],
```

Comment:
I love this.  reallllly elegant and simple solution to test what we're trying to test!

Reply:
Hmm, actually just noticed a bug in it. We should have base be the same as the requested page. That way a subsequent request for page 1 after a request for page 2 will actually return the items.

Reply:
haha, well I stand corrected then!  but either way I like the concept!

Reply:
Alright, modified. Now it's properly non-stateful.